openwisp_firmware_upgrader/tests/test_admin.py

```diff
@@ -571,8 +572,14 @@ def test_upgrade_related(self, *args):
                 },
                 follow=True,
             )
+            id_attr = (
+                ' id="id_upgrade_options_error"' if django.VERSION >= (5, 2) else ''
+            )
             self.assertContains(
-                response, '<ul class="errorlist"><li>Enter a valid JSON.</li></ul>'
+                response,
+                ('<ul class="errorlist"{}><li>Enter a valid JSON.</li></ul>').format(
+                    id_attr
+                ),
```

Comment:
We can use f-strings here which have a more concise syntax.
```suggestion
                response, f'<ul class="errorlist"{id_attr}><li>Enter a valid JSON.</li></ul>'
```